jetty-client/src/main/java/org/eclipse/jetty/client/AuthenticationProtocolHandler.java

```diff
@@ -241,8 +241,13 @@ private void forwardFailureComplete(HttpRequest request, Throwable requestFailur
                 if (matcher.matches())
                 {
                     String type = matcher.group(1);
-                    String realm = matcher.group(2);
-                    String params = matcher.group(3);
+                    String realm = matcher.group(3);
+                    String params;
+                    if(matcher.group(2) != null)
```

Comment:
Put a space between `if` and `(`.
Introduce a local variable for `matcher.group(2)`, maybe named `beforeRealm`.